### ISSUE TITLE
commit_pending --all exits immediately after error

### ISSUE THREAD

**Issue by jmaxxz** [WeblateOrg/weblate#1528]
### Steps to reproduce
1. Create multiple projects/components
2. Create a error on one of the components (unmerged files, etc)
3. Create pending changes on all other components
4. Run `./manage.py commit_pending --all`

### Actual behaviour
When the error is encountered the process quits and no other components are are processed.

### Expected behaviour
Upon error manage should continue on to the next component. As `./manage.py commit_pending --all` is often run from a non interactive shell (cron) and to ensure all components stay up-to-date. Similarly behavior should be expected for updategit, pushgit, and all other management operations which run on a per component basis.

**Comment by nijel** [comment]
Can you please clarify which error did cause this? I can't find any such examples on my servers...

**Comment by jmaxxz** [comment]
I don't have that information in my records anymore.

**Comment by jmaxxz** [comment]
Similiar issue:
```
 ./manage.py pushgit --all
```

One component out of 82 had issue, this resulted in the process bailing out and the rest of the components not being pushed.
```
Traceback (most recent call last):
  File "./manage.py", line 31, in <module>
    execute_from_command_line(sys.argv)
  File "/usr/local/lib/python2.7/dist-packages/django/core/management/__init__.py", line 363, in execute_from_command_line
    utility.execute()
  File "/usr/local/lib/python2.7/dist-packages/django/core/management/__init__.py", line 355, in execute
    self.fetch_command(subcommand).run_from_argv(self.argv)
  File "/usr/local/lib/python2.7/dist-packages/django/core/management/base.py", line 283, in run_from_argv
    self.execute(*args, **cmd_options)
  File "/usr/share/weblate/weblate/trans/management/commands/__init__.py", line 42, in execute
    super(WeblateCommand, self).execute(*args, **options)
  File "/usr/local/lib/python2.7/dist-packages/django/core/management/base.py", line 330, in execute
    output = self.handle(*args, **options)
  File "/usr/share/weblate/weblate/trans/management/commands/pushgit.py", line 39, in handle
    subproject.do_push(None, force_commit=options['force_commit'])
  File "/usr/share/weblate/weblate/trans/models/subproject.py", line 112, in wrapper
    return func(self, *args, **kwargs)
  File "/usr/share/weblate/weblate/trans/models/subproject.py", line 779, in do_push
    self.do_update(request)
  File "/usr/share/weblate/weblate/trans/models/subproject.py", line 112, in wrapper
    return func(self, *args, **kwargs)
  File "/usr/share/weblate/weblate/trans/models/subproject.py", line 725, in do_update
    ret = self.update_branch(request, method=method)
  File "/usr/share/weblate/weblate/trans/models/subproject.py", line 112, in wrapper
    return func(self, *args, **kwargs)
  File "/usr/share/weblate/weblate/trans/models/subproject.py", line 979, in update_branch
    method(abort=True)
  File "/usr/share/weblate/weblate/trans/vcs.py", line 444, in merge
    self.execute(['merge', '--abort'])
  File "/usr/share/weblate/weblate/trans/vcs.py", line 194, in execute
    self.last_output = self._popen(args, self.path)
  File "/usr/share/weblate/weblate/trans/vcs.py", line 181, in _popen
    output.decode('utf-8')
weblate.trans.vcs.RepositoryException: fatal: There is no merge to abort (MERGE_HEAD missing). (128)

```

**Comment by nijel** [comment]
This is bug in the repository handling, which I've just fixed in 9060662f56c40c7f5b9e7ec910d5b6e72fb1ff81.

**Comment by nijel** [comment]
Thank you for your report, the issue you have reported has just been fixed. 

* In case you see problem with the fix, please comment on this issue. 
* In case you see similar problem, please open separate issue.
* If you are happy with the outcome, consider supporting Weblate by [donating](https://weblate.org/donate/).
